src/layouts/Root/components/SearchInput.tsx

```diff
@@ -0,0 +1,186 @@
+
+          & .algolia-docsearch-suggestion--highlight {
+            background-color: ${highlightBackgroundColor};
+            padding: 0;
+            color: inherit;
+          }
+
+          & .algolia-docsearch-suggestion--category-header {
+            margin: 0;
+            margin-bottom: ${theme.space.xxs};
+            border-color: ${separatorColor};
+            padding: ${theme.space.xs} ${theme.space.sm} ${theme.space.sm};
+            color: inherit;
+            font-weight: ${theme.fontWeights.semibold};
+
+            & .algolia-docsearch-suggestion--highlight {
+              box-shadow: none;
+              background-color: transparent;
+            }
+          }
+
+          & .algolia-docsearch-suggestion--subcategory-column {
+            padding: 0;
+            padding-right: ${theme.space.sm};
+            color: ${metaColor};
+            font-size: inherit;
+
+            & .algolia-docsearch-suggestion--highlight {
+              background-color: transparent;
+            }
+          }
+
+          & .algolia-docsearch-suggestion--content {
+            padding: 0;
+            padding-left: ${theme.space.sm};
+          }
+
+          & .algolia-docsearch-suggestion--text {
+            line-height: ${theme.lineHeights.sm};
+            color: ${metaColor};
+            font-size: ${theme.fontSizes.sm};
+
+            & .algolia-docsearch-suggestion--highlight {
+              box-shadow: none;
+              background-color: transparent;
+              text-decoration: underline;
+            }
+          }
+
+          & .algolia-docsearch-suggestion--title {
+            margin: 0;
+            font-size: inherit;
+            font-weight: ${theme.fontWeights.regular};
+          }
+
+          & .algolia-docsearch-suggestion--wrapper {
+            padding: ${theme.space.xs} ${theme.space.base * 9}px;
+
+            & > *::before {
+              background-color: ${separatorColor};
+            }
+          }
+        }
+
+        & .ds-cursor {
+          & .algolia-docsearch-suggestion--wrapper {
+            background-color: ${hoverBackgroundColor};
+
+            & > *::before {
+              background-color: ${hoverSeparatorColor};
+            }
+
+            & .algolia-docsearch-suggestion--content {
+              background-color: transparent;
+            }
+          }
+        }
+
+        & .algolia-docsearch-footer {
+          margin: ${theme.space.xs} ${theme.space.sm};
+          width: 120px;
+        }
+      }
+    }
+  `;
+};
+
+declare global {
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  interface Window {
+    docsearch: any;
+  }
+}
+
+export const SearchInput = React.forwardRef<HTMLInputElement, HTMLAttributes<HTMLInputElement>>(
+  ({ id, ...props }, ref) => {
+    useEffect(() => {
+      if (typeof window !== 'undefined' /* not SSR */ && window.docsearch) {
+        window.docsearch({
+          apiKey: 'c1ab842b2bfd6f3f7104a8e903c615bf',
```

Comment:
Any reason to env variable this in Netlify?

Reply:
Nah, this is how the cool kids [do it](https://github.com/reactjs/reactjs.org/blob/master/src/components/LayoutHeader/DocSearch.js). It's a read-only key, so no worries about leaked index security.